File: src/main/java/com/jacuum/engine/ActiveMessaging.java
package com.jacuum.engine;

import org.springframework.messaging.simp.SimpMessagingTemplate;

public final class ActiveMessaging implements Messaging {
    private final SimpMessagingTemplate delegate;

    public ActiveMessaging(final SimpMessagingTemplate delegate) {
        this.delegate = delegate;
    }

    @Override
    public void send(final String destination, final Object payload) {
        this.delegate.convertAndSend(destination, payload);
    }
}

File: src/main/kotlin/com/jacuum/engine/ActiveMessaging.kt
package com.jacuum.engine

import org.springframework.messaging.simp.SimpMessagingTemplate

class ActiveMessaging(private val delegate: SimpMessagingTemplate) : Messaging {
    override fun send(destination: String, payload: Any) {
        delegate.convertAndSend(destination, payload)
    }
}
